iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/CalculatedBusImpl.java

```diff
@@ -26,9 +24,38 @@
 
     private final List<NodeTerminal> terminals;
 
-    CalculatedBusImpl(String id, VoltageLevelExt voltageLevel, List<NodeTerminal> terminals) {
+    private NodeTerminal terminalRef;
+
+    CalculatedBusImpl(String id, NodeBreakerVoltageLevel voltageLevel, TIntArrayList nodes, List<NodeTerminal> terminals) {
         super(id, voltageLevel);
         this.terminals = Objects.requireNonNull(terminals);
+        this.terminalRef = findTerminal(voltageLevel, nodes, terminals);
+    }
+
+    private static NodeTerminal findTerminal(NodeBreakerVoltageLevel voltageLevel, TIntArrayList nodes, List<NodeTerminal> terminals) {
+        if (!terminals.isEmpty()) {
+            return terminals.get(0);
+        }
+
+        NodeTerminal[] terminal = new NodeTerminal[1];
```

Comment:
Maybe used a reference type instead? (`AtomicReference` or maybe `WeakReference`)
It looks clearer to me than the array-based idiom, but it's subjective.

Reply:
Personally, I like this way to create mutable reference using a single element array.
Problem with AtomicReference is that it involves an unnecessary lock.

Reply:
Ok.
Maybe some day we should just have a silly in-house class for this use case, like `Ref<T>`. The array stuff may look surprising to people who have never seen it before. Whatever ... :)